libminifi/src/core/ProcessSession.cpp

```diff
@@ -296,7 +296,7 @@ void ProcessSession::append(const std::shared_ptr<core::FlowFile> &flow, const i
     size_t stream_size_before_callback = stream->size();
     // this prevents an issue if we write, above, with zero length.
     if (stream_size_before_callback > 0)
-      stream->seek(stream_size_before_callback + 1);
+      stream->seek(stream_size_before_callback);
```

Comment:
yes, unclear why we seek to `size + 1` instead of `size`, but it has been like this since the beginning, since `BufferStream::seek` does not affect write only read it was basically noop while with `FileStream` this is a problem